simple/stats/events_importer.py

```diff
@@ -0,0 +1,280 @@
+      self.entity_type = self._resolve_entity_type()
+      if self.entity_type:
+        logging.info("Resolved entity type: %s", self.entity_type)
+    if not self.entity_type:
+      logging.warning(
+          "Could not resolve entity type. Event triples will not be associated with any entity."
+      )
+
+    triples: list[Triple] = []
+    for index, row in self.df.iterrows():
+      # If id column is configured, use it as the event dcid else generate based on row index.
+      dcid = row[
+          id_column_name] if id_column_name else f"{self.event_type}_{index}"
+
+      entity = row.iloc[0]
+      date = row.iloc[1]
+      properties: dict[str, str] = {}
+
+      for i, (k, v) in enumerate(row.items()):
+        if i < 2:
+          continue
+        properties[k] = v
+
+      event = Event(dcid,
+                    self.event_type,
+                    entity_type=self.entity_type,
+                    entity=entity,
+                    date=date,
+                    provenance_id=self.provenance,
+                    properties=properties)
+      triples.extend(event.triples())
+
+      self.db.insert_triples(triples)
+
+  def _resolve_entity_type(self) -> str:
+    all_entity_dcids = self.df.iloc[:, 0].tolist()
+    sample_entity_dcids = random.sample(
+        all_entity_dcids,
+        min(len(all_entity_dcids), _SAMPLE_ENTITY_RESOLUTION_SIZE))
+    logging.info("Resolving entity type from sample entities: %s",
+                 sample_entity_dcids)
+    return dc.resolve_entity_type(sample_entity_dcids)
+
+  def _resolve_entities(self) -> None:
+    df = self.df
+    # get first (0th) column
+    column = df.iloc[:, 0]
+
+    pre_resolved_entities = {}
+
+    def remove_pre_resolved(entity: str) -> bool:
+      if entity.startswith(constants.DCID_OVERRIDE_PREFIX):
+        pre_resolved_entities[entity] = entity[
+            len(constants.DCID_OVERRIDE_PREFIX):].strip()
+        return False
+      return True
+
+    entities = list(filter(remove_pre_resolved, column.tolist()))
+
+    logging.info("Found %s entities pre-resolved.", len(pre_resolved_entities))
+
+    logging.info("Resolving %s entities of type %s.", len(entities),
+                 self.entity_type)
+    dcids = self._resolve(entity_column_name=df.columns[0], entities=entities)
+    logging.info("Resolved %s of %s entities.", len(dcids), len(entities))
+
+    # Replace resolved entities.
+    column.replace(dcids, inplace=True)
+    unresolved = set(entities).difference(set(dcids.keys()))
+    unresolved_list = sorted(list(unresolved))
+
+    # Replace pre-resolved entities without the "dcid:" prefix.
+    column.replace(pre_resolved_entities, inplace=True)
+
+    if unresolved_list:
+      logging.warning("# unresolved entities which will be dropped: %s",
+                      len(unresolved_list))
+      logging.warning("Dropped entities: %s", unresolved_list)
+      df.drop(df[df.iloc[:, 0].isin(values=unresolved_list)].index,
```

Comment:
nit: df = df[df['dcid'].isin(unresolved_list)]

Reply:
Ack - will do in a separate PR (see comment above).